src/transformers/trainer.py

```diff
@@ -1253,18 +1253,36 @@ def torch_jit_model_eval(self, model, dataloader, training=False):
                 return model
             jit_inputs = []
             example_batch = next(iter(dataloader))
+            if "labels" in example_batch:
+                if self.label_smoother is not None:
+                    example_batch.pop("labels")
+                elif version.parse(version.parse(torch.__version__).base_version) < version.parse("1.14.0"):
```

Comment:
This mean the whole eval will fail in 99.99% of the case when users do not use label smoothing (it's a training technique)

Reply:
the fix of jit trace failure is https://github.com/pytorch/pytorch/pull/81623, I add it to the doc to let user know it, which is expected to be merged in pytorch1.14.0. However, we will catch the trace failure issue and fallback to the original path